tree_arena/src/tree_arena_unsafe.rs

```diff
@@ -276,6 +286,64 @@ impl<T> TreeArena<T> {
     pub fn get_id_path(&self, id: impl Into<NodeId>) -> Vec<NodeId> {
         self.data_map.get_id_path(id.into(), None)
     }
+
+    // Moves the given child (along with all children) to the new parent.
+    //
+    /// # Panics
+    ///
+    /// If the parent is actually a child of the to-be-reparented node.
+    /// If the to-be-reparented node is a root node.
+    /// If both given IDs are equal.
+    pub fn reparent(&mut self, child: impl Into<NodeId>, new_parent: impl Into<NodeId>) {
+        let child_id = child.into();
+        let new_parent_id = new_parent.into();
+
+        assert_ne!(
+            child_id, new_parent_id,
+            "expected child to be different from new_parent"
```

Comment:
```suggestion
            "expected child ({child_id}) to be different from new_parent ({new_parent_id})"
```

As a way to give a bit more context. It doesn't add all that much value, but it shouldn't really hurt either.
This applies to most of the asserts added here